src/physlr-molecules.cc

```diff
@@ -323,6 +329,78 @@ biconnectedComponents(graph_t& subgraph, vertexToComponent_t& vertexToComponent)
 		}
 		++moleculeNum;
 	}
+	return moleculeNum;
+}
+
+void
+bin_components(
+    componentToVertexSet_t& source,
+    componentToVertexSet_t& binned_neighbours,
+    uint64_t bin_size = 50)
+{
+	// //   Iterate over each component and if its bigger than bin_size:
+	// //   randomly split the component (set of vertices) into smaller even bins
+
+	std::vector<uint64_t> components_size;
+	uint64_t neighborhood_size;
+	uint64_t components_count;
+	for (uint64_t i = 0; i < source.size(); i++) { // NOLINT
+		neighborhood_size = source[i].size();
+		components_count = ((neighborhood_size - 1) / bin_size) + 1;
+		components_size.push_back(components_count);
+	}
```

Comment:
You can use the range-based for loop here

Reply:
Does that loop access the elements in the same order this loop does? as I am pushing some values for each element into `components_size` and I later use them together so I need them to be consistent. I previously experienced inconsistency using the range-based for loop (when I tried to simulate the `i` counter for the other loop which I asked about yesterday)...

Reply:
Oh I see. Stick to this then. Might want to add a comment saying specifically you need it because of order in case someone in the future decides to change the code here.